File: bannerlayout/src/main/java/com/wkz/bannerlayout/imagemanager/GlideImageManager.java
package com.wkz.bannerlayout.imagemanager;

import android.support.annotation.NonNull;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.engine.DiskCacheStrategy;
import com.bumptech.glide.request.RequestOptions;
import com.wkz.bannerlayout.listener.BannerModelCallBack;
import com.wkz.bannerlayout.listener.ImageDisplayManager;
import com.wkz.bannerlayout.widget.BannerDefaults;

public class GlideImageManager implements ImageDisplayManager {

    private RequestOptions requestOptions;

    @NonNull
    public ImageView display(@NonNull ViewGroup container, @NonNull BannerModelCallBack model) {
        ImageView imageView = new ImageView(container.getContext());
        Glide.with(imageView.getContext())
                .applyDefaultRequestOptions(
                        this.requestOptions
                                .placeholder(BannerDefaults.GLIDE_PLACEHOLDER_DRAWABLE)
                                .error(BannerDefaults.GLIDE_ERROR_DRAWABLE)
                                .fallback(BannerDefaults.GLIDE_ERROR_DRAWABLE)
                                .diskCacheStrategy(DiskCacheStrategy.ALL)
                )
                .load(model.getBannerUrl())
                .into(imageView);
        return imageView;
    }

    public GlideImageManager() {
        this.requestOptions = (new RequestOptions()).centerCrop();
    }
}

File: bannerlayout/src/main/java/com/fphoenixcorneae/bannerlayout/imagemanager/GlideImageManager.kt
package com.fphoenixcorneae.bannerlayout.imagemanager

import android.view.ViewGroup
import android.widget.ImageView
import com.bumptech.glide.Glide
import com.bumptech.glide.load.engine.DiskCacheStrategy
import com.bumptech.glide.load.resource.bitmap.CenterCrop
import com.bumptech.glide.load.resource.bitmap.RoundedCorners
import com.bumptech.glide.request.RequestOptions
import com.fphoenixcorneae.bannerlayout.listener.BannerModelCallBack
import com.fphoenixcorneae.bannerlayout.listener.ImageDisplayManager
import com.fphoenixcorneae.bannerlayout.widget.BannerDefaults

class GlideImageManager(cornerRadius: Int) : ImageDisplayManager {
    private val requestOptions: RequestOptions = RequestOptions().transform(
            CenterCrop(),
            RoundedCorners(cornerRadius)
    )
            .placeholder(BannerDefaults.GLIDE_PLACEHOLDER_DRAWABLE)
            .error(BannerDefaults.GLIDE_ERROR_DRAWABLE)
            .fallback(BannerDefaults.GLIDE_ERROR_DRAWABLE)
            .diskCacheStrategy(DiskCacheStrategy.ALL)

    override fun display(container: ViewGroup, model: BannerModelCallBack<*>?): ImageView {
        val imageView = ImageView(container.context)
        Glide.with(imageView.context)
                .applyDefaultRequestOptions(requestOptions)
                .load(model?.bannerUrl)
                .into(imageView)
        return imageView
    }
}